pkg/operator/ceph/object/spec.go

```diff
@@ -280,17 +280,34 @@ func (c *clusterConfig) makeRGWPodSpec(rgwConfig *rgwConfig) (v1.PodTemplateSpec
 }
 
 func (c *clusterConfig) createCaBundleUpdateInitContainer(rgwConfig *rgwConfig) v1.Container {
-	caBundleMount := v1.VolumeMount{Name: caBundleVolumeName, MountPath: caBundleSourceCustomDir, ReadOnly: true}
+	caBundleMount := v1.VolumeMount{Name: caBundleVolumeName, MountPath: "/tmp/ca-bundle", ReadOnly: true}
 	volumeMounts := append(controller.DaemonVolumeMounts(c.DataPathMap, rgwConfig.ResourceName, c.clusterSpec.DataDirHostPath), caBundleMount)
 	updatedCaBundleDir := "/tmp/new-ca-bundle/"
 	updatedBundleMount := v1.VolumeMount{Name: caBundleUpdatedVolumeName, MountPath: updatedCaBundleDir, ReadOnly: false}
 	volumeMounts = append(volumeMounts, updatedBundleMount)
 	return v1.Container{
 		Name:    "update-ca-bundle-initcontainer",
 		Command: []string{"/bin/bash", "-c"},
-		// copy all content of caBundleExtractedDir to avoid directory mount itself
 		Args: []string{
-			fmt.Sprintf("/usr/bin/update-ca-trust extract; cp -rf %s/* %s", caBundleExtractedDir, updatedCaBundleDir),
+			fmt.Sprintf(`
+				OS_ID=$(grep '^ID=' /etc/os-release | cut -d'=' -f2 | tr -d '"')
+				OS_ID_LIKE=$(grep '^ID_LIKE=' /etc/os-release | cut -d'=' -f2 | tr -d '"')
+
+				if [[ "$OS_ID" == "sles" || "$OS_ID" == "opensuse" || "$OS_ID_LIKE" == *"suse"* ]]; then
+					CABUNDLE="/etc/ssl/ca-bundle.pem"
+				elif [[ "$OS_ID" == "debian" || "$OS_ID" == "ubuntu" || "$OS_ID_LIKE" == *"debian"* ]]; then
+					CABUNDLE="/etc/ssl/certs/ca-certificates.crt"
+				else
+					CABUNDLE="/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
+				fi
+
+				if [ ! -f "$CABUNDLE" ]; then
+					echo "CA bundle not found at $CABUNDLE!" >&2
+					exit
+				fi
+
+				cat "$CABUNDLE" /tmp/ca-bundle/custom-ca-bundle.crt > %s/combined-ca-bundle.crt
```

Comment:
I'm a little nervous about changing the implementation here for the upstream Ceph images that have `update-ca-trust` available. IMO, the existing `/usr/bin/update-ca-trust extract; cp -rf ...` commands should continue to be the same so that we don't have any unexpected regressions.

I would expect fedora/rhel/centos to continue to use `update-ca-trust extract` instead of `cat`.

Reply:
hmm, in that case we can use commands for suse and debian as well.

Reply:
That's fine with me. I don't know what those commands are, but I imagine it's generally better to use the commands that are available.

Reply:
done